src/lib/y2packager/product_spec_readers/control.rb

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2packager/control_product_spec"
+
+Yast.import "Arch"
+Yast.import "Linuxrc"
+Yast.import "ProductFeatures"
+
+module Y2Packager
+  module ProductSpecReaders
+    # Reads product specifications from the control file
+    class Control
+      include Yast::Logger
+
+      # map the Arch.architecture to the arch expected by SCC
+      REG_ARCH = {
+        "s390_32" => "s390",
+        "s390_64" => "s390x",
+        # ppc64le is the only supported PPC arch, we do not have to distinguish the BE/LE variants
+        "ppc64"   => "ppc64le"
```

Comment:
I am a bit suspicious that it is the first time we need such mapping in YaST. So I expect it has to live also somewhere else.

Reply:
Yes, I will add a note to address it in a near future. It was taken [from here](https://github.com/yast/yast-yast2/blob/4f0560a55f0b4ece41ed50063fd7baf00cd8eedc/library/packages/src/lib/y2packager/product_control_product.rb#L35-L40).